src/app/OperationalDeviceProxy.h

```diff
@@ -99,33 +110,29 @@ class DLL_EXPORT OperationalDeviceProxy : public DeviceProxy, public SessionEsta
      */
     void OnConnectionExpired(SessionHandle session) override;
 
+    void OnNodeIdResolved(const Dnssd::ResolvedNodeData * nodeResolutionData)
+    {
+        if (nodeResolutionData != nullptr)
```

Comment:
is there any value in allowing null node resolution? How about making the argument a reference and have the callers check for null?